lib/proxy_api/external_ipam.rb

```diff
@@ -0,0 +1,140 @@
+module ProxyAPI
+  class ExternalIpam < ProxyAPI::Resource
+    def initialize(args)
+      @url = args[:url] + "/ipam"
+      super args
+    end
+
+    # Retrieves the next available IP address for the specified subnet
+    #
+    # Inputs: 1. subnet(object): The Foreman subnet
+    #         2. mac: The mac address of the interface obtaining the IP address for
+    # Returns: Hash with "next_ip", or hash with "error"
+    # Examples:
+    #   Response if success:
+    #     {"cidr":"100.20.20.0/24","next_ip":"100.20.20.11"}
+    #   Response if error:
+    #     {"error":"The specified subnet does not exist in external ipam."}
+    def next_ip(subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      get("/next_ip?cidr=#{cidr}")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to retrieve the next available IP for subnet %s from External IPAM."), cidr)
+    end
+
+    # Adds an IP address to the specified subnet
+    #
+    # Inputs: 1. ip(string). IP address to be added.
+    #         2. subnet(object): The Foreman subnet
+    # Returns: Hash with "message" on success, or hash with "error"
+    # Examples:
+    #   Response if success:
+    #     {"message":"IP 100.10.10.123 added to subnet 100.10.10.0/24 successfully."}
+    #   Response if error:
+    #     {"error":"The specified subnet does not exist in external ipam."}
+    def add_ip_to_subnet(ip, subnet)
+      cidr = subnet.network + '/' + subnet.cidr.to_s if subnet.network.present?
+      parse(post({:ip => ip, :cidr => cidr}, "/add_ip_to_subnet"))
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to add IP %s to the subnet %s in External IPAM."), ip, cidr)
+    end
+
+    # Get a list of sections from external ipam
+    #
+    # Input: None
+    # Returns: An array of sections on success, hash with "error" key otherwise
+    # Examples
+    #   Response if success: [
+    #     {"id":"4","name":"Test Section","description":"A Test Section"},
+    #     {"id":"5","name":"Awesome Section","description":"A totally awesome Section"}
+    #   ]
+    #   Response if error:
+    #     {"error":"Unable to connect to external ipam server"}
+    def get_sections
+      get("/sections")
+    rescue => e
+      raise ProxyException.new(url, e, N_("Unable to obtain sections from External IPAM."))
+    end
```

Comment:
Still not sold about database "IDs", I assume "name" is unique in MyIPAM then use that to find sections. There will be implementations of IPAM without database I am pretty sure.